core/domain/stats_domain_test.py

```diff
@@ -1085,6 +1085,10 @@ def test_answer_group_index_must_be_positive(self):
             self.submitted_answer,
             'Expected answer_group_index to be non-negative')
 
+    def test_rule_spec_index_can_be_none(self):
+        self.submitted_answer.rule_spec_index = None
```

Comment:
This test should be stronger -- try empty string etc.

Reply:
Done.